### PR TITLE
Raise error on product form if 2 variations have the same name + value

### PR DESCRIPTION
**Pull request by ithiame**
🦋 Raise error on product form if 2 variations have the same name + value #1515

### CODE REVIEW

File: src/lib/components/ProductForm.svelte [hunk 1]
```diff
@@ -126,7 +128,26 @@
 			} else {
 				priceAmountElement.setCustomValidity('');
 			}
+			let seen = new Set();
```
Comment (coyotte508) [inline src/lib/components/ProductForm.svelte:131]: use `const`, and type it appropriately, `new Set<string>()`

File: src/lib/components/ProductForm.svelte [hunk 2]
```diff
@@ -126,7 +128,26 @@
...
+			for (let i = 0; i < variationLabelsValues.length; i++) {
```
Comment (coyotte508) [inline src/lib/components/ProductForm.svelte:132]: use `for const of`

File: src/lib/components/ProductForm.svelte [hunk 3]
```diff
@@ -126,7 +128,26 @@
...
+				const key = `${(
+					variationLabelsNames[i] ||
+					product.variations?.[i].name ||
+					''
+				).toLowerCase()}-${(
+					variationLabelsValues[i] ||
+					product.variations?.[i].value ||
+					''
+				).toLowerCase()}`;
```
Comment (coyotte508) [inline src/lib/components/ProductForm.svelte:141]: Use `JSON.stringify({name, value})` - that way if I have a variation with name "a-b" and val "c", and another with "a" and "b-c", it won't cause issues

**Review by coyotte508 — APPROVED** [review]

**Comment by Tirodem** [comment]
image

**Pull request merged**